tracer/src/Datadog.Trace/FaultTolerant/FaultTolerantNativeMethods.cs

```diff
@@ -0,0 +1,82 @@
+using Datadog.Trace.Debugger.Sink.Models;
+
+namespace Datadog.Trace.FaultTolerant
+{
+    // ReSharper disable MemberHidesStaticFromOuterClass
+    internal static class FaultTolerantNativeMethods
+    {
+        private static readonly bool IsWindows = FrameworkDescription.Instance.IsWindows();
+
+        public static bool ShouldHeal(IntPtr moduleId, int methodToken, string instrumentationVersion, int products)
+        {
+            if (IsWindows)
+            {
+                return Windows.ShouldHeal(moduleId, methodToken, instrumentationVersion, products);
+            }
+            else
+            {
+                return NonWindows.ShouldHeal(moduleId, methodToken, instrumentationVersion, products);
+            }
+        }
+
+        public static void ReportSuccessfulInstrumentation(IntPtr moduleId, int methodToken, string instrumentationVersion, int products)
+        {
+            if (IsWindows)
+            {
+                Windows.ReportSuccessfulInstrumentation(moduleId, methodToken, instrumentationVersion, products);
+            }
+            else
+            {
+                NonWindows.ReportSuccessfulInstrumentation(moduleId, methodToken, instrumentationVersion, products);
+            }
+        }
+
+        // the "dll" extension is required on .NET Framework
+        // and optional on .NET Core
+        // These methods are rewritten by the native tracer to use the correct paths
+        private static class Windows
+        {
+            [DllImport("Datadog.Tracer.Native.dll")]
+            public static extern bool ShouldHeal(
+                [In] IntPtr moduleId,
+                [In] int methodToken,
+                [In] [MarshalAs(UnmanagedType.LPWStr)] string instrumentationVersion,
+                [In] int products);
+
+            [DllImport("Datadog.Tracer.Native.dll")]
+            public static extern void ReportSuccessfulInstrumentation(
+                [In] IntPtr moduleId,
+                [In] int methodToken,
+                [In] [MarshalAs(UnmanagedType.LPWStr)] string instrumentationVersion,
+                [In] int products);
+        }
+
+        // assume .NET Core if not running on Windows
+        // These methods are rewritten by the native tracer to use the correct paths
+        private static class NonWindows
+        {
+            [DllImport("Datadog.Tracer.Native")]
+            public static extern bool ShouldHeal(
+                [In] IntPtr moduleId,
+                [In] int methodToken,
+                [In] [MarshalAs(UnmanagedType.LPWStr)] string instrumentationVersion,
+                [In] int products);
+
+            [DllImport("Datadog.Tracer.Native")]
+            public static extern void ReportSuccessfulInstrumentation(
+                [In] IntPtr moduleId,
+                [In] int methodToken,
+                [In][MarshalAs(UnmanagedType.LPWStr)] string instrumentationVersion,
+                [In] int products);
```

Comment:
Since you rewrite the PInvokeMaps, do you still need to have one Windows and one Non-Windows?
Since the signatures are identicals, you can have only one static class, no ?
(we did this for the profiler)